manager/sinks.go

```diff
@@ -22,9 +22,9 @@ import (
 	sink_api "github.com/GoogleCloudPlatform/heapster/sinks/api/v1"
 )
 
-func newSinks() ([]sink_api.ExternalSink, error) {
+func newSinks(sinkUris []string) ([]sink_api.ExternalSink, error) {
```

Comment:
Hadn't noticed that you were describing the same struct in two different ways. What about it being some sort of interface with the underlying type being the actual config, be it `GCMConfig` or whatever?

The only problem then would be the json key names.